front/pages/api/w/[wId]/data_sources/managed.ts

```diff
@@ -0,0 +1,212 @@
+            splitter_id: "base_v0",
+            max_chunk_size: dataSourceMaxChunkSize,
+            use_cache: false,
+          },
+          credentials,
+        }
+      );
+
+      if (dustDataSource.isErr()) {
+        return apiError(req, res, {
+          status_code: 500,
+          api_error: {
+            type: "data_source_error",
+            message: `Could not create the data source. reason: ${dustDataSource.error}`,
+          },
+        });
+      }
+
+      const dataSource = await DataSource.create({
+        name: dataSourceName,
+        description: dataSourceDescription,
+        //assuming managed data sources are always private for now
+        visibility: "private",
+        config: JSON.stringify(dustDataSource.value.data_source.config),
+        dustAPIProjectId: dustProject.value.project.project_id.toString(),
+        workspaceId: owner.id,
+      });
+      const systemAPIKeyRes = await getSystemApiKey(owner.id);
+      if (systemAPIKeyRes.isErr()) {
+        console.error(
+          `Could not create the system API key: ${systemAPIKeyRes.error}`
+        );
+        return apiError(req, res, {
+          status_code: 500,
+          api_error: {
+            type: "internal_server_error",
+            message: "Could not create the system API key",
+          },
+        });
+      }
+
+      try {
+        const connectorsRes = await fetch(
+          `${CONNECTORS_API_URL}/connectors/create/slack`,
+          {
+            method: "POST",
+            headers: {
+              "Content-Type": "application/json",
+            },
+            body: JSON.stringify({
+              workspaceId: owner.id.toString(),
+              APIKey: systemAPIKeyRes.value.secret,
+              dataSourceName: dataSource.name,
+              nangoConnectionId: req.body.nangoConnectionId,
+            }),
+          }
+        );
+        if (!connectorsRes.ok) {
+          return apiError(req, res, {
+            status_code: 500,
+            api_error: {
+              type: "internal_server_error",
+              message: "Failed to create the connector.",
+            },
+          });
+        }
+        const connectorsResJson = (await connectorsRes.json()) as {
+          connectorId: string;
+        };
+
+        dataSource.connectorId = connectorsResJson.connectorId;
+        dataSource.connectorProvider = "slack";
+        dataSource.save();
+
+        res.redirect(`/${owner.sId}/ds/${dataSource.name}`);
+        return;
+      } catch (err) {
+        return apiError(req, res, {
+          status_code: 500,
+          api_error: {
+            type: "internal_server_error",
+            // Do we want to forward the message from the connector service here?
+            message: "Failed to create the connector.",
+          },
+        });
+      }
+
+    default:
+  }
+}
+
+// Not sure where this should go.
+// Lets find out at code review time.
+async function getSystemApiKey(
```

Comment:
should we share some of that code with the POST `keys` endpoint ? Maybe we could have some `business_logic` (or whatever other name) where we have re-usable functions cc @spolu, could also be useful for the code duplication we have between `/api` and `/v1`

Reply:
currently we have `lib/api/*` that serves that purpose to some extent which is `front` server-side only (vs the rest of the function that are used both server-side and in the front-end

We can rename that `lib/api/` to something else of course

Reply:
But for now I would be fine with this living in `lib/api/system_keys.ts` or `lib/system_keys.ts` we can figure out the naming part later.

Reply:
`lib` sgtm

Reply:
I think we will need it in other endpoints so I moved it to `lib/auth.ts`